dev/sparktestsupport/shellutils.py

```diff
@@ -57,7 +55,12 @@ def run_cmd(cmd, return_output=False):
         if return_output:
             return subprocess_check_output(cmd).decode(sys.getdefaultencoding())
         else:
-            return subprocess_check_call(cmd)
+            popen = subprocess.Popen(cmd, stdout=subprocess.PIPE)
+            for stdout_line in popen.stdout:
+                print(stdout_line.decode(sys.getdefaultencoding()), end='')
+            popen.stdout.close()
+            return_code = popen.wait()
+            return return_code
```

Comment:
Hm, @shaneknapp btw why do we need to do this? at least it seems going to work differently comparing to the past:

> If the return code was zero then return, otherwise raise CalledProcessError. The CalledProcessError object will have the return code in the returncode attribute.

https://docs.python.org/2/library/subprocess.html#subprocess.check_call

Reply:
`subprocess.check_call()` returns the exit code of the process, so i'm literally mimicking previous functionality (in a slightly more verbose way).

Reply:
i will also admit that there's a bit of `stackoverflow` in that bit of code.  ;)

Reply:
@HyukjinKwon ok, yeah.  i see what you mean.  let me see if i can get `subprocess.run()` with `check=True` to work...

Reply:
looks like https://github.com/apache/spark/pull/26330/commits/48e318b9aa41e90f1768e5b6a583215a0cbb513a is working as intended (https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/113467/console)